--- /usr/local/lib/python3.12/site-packages/ray/rllib/models/torch/torch_action_dist.py	2026-07-01 10:52:09.534385771 +0000
+++ /usr/local/lib/python3.12/site-packages/ray/rllib/models/torch/torch_action_dist_new.py	2026-07-01 10:50:24.735538186 +0000 
@@ -613,7 +613,7 @@
         See issue #4440 for more details.
         """
         self.epsilon = torch.tensor(1e-7).to(inputs.device)
-        concentration = torch.exp(inputs) + self.epsilon
+        concentration = inputs + self.epsilon
         self.dist = torch.distributions.dirichlet.Dirichlet(
             concentration=concentration,
             validate_args=True,
@@ -622,7 +622,9 @@
 
     @override(ActionDistribution)
     def deterministic_sample(self) -> TensorType:
-        self.last_sample = nn.functional.softmax(self.dist.concentration, dim=-1)
+        # Ray RLLib original: nn.functional.softmax(self.dist.concentration, dim=-1)o
+        # We replaced with the true mean of the distribution as defined for.
+        self.last_sample = self.dist.concentration / self.dist.concentration.sum(dim=-1, keepdim=True)
         return self.last_sample
 
     @override(ActionDistribution)
